### ISSUE TITLE
Shimmer 0.5.0 Console not working properly (at least according to docs)

### ISSUE THREAD

**Issue by kmhughes** [openmhealth/shimmer#90]
Hi folks,

I am trying to use Shimmer 0.5.0 because I want the data endpoints support that weren't supported in earlier versions.

The console doesn't appear to work, for example I see nothing in the settings window.  Checking with the Chrome debugger and looking at the Java code, it seems some calls to the resource server, such as /registry no longer work. I also cannot see any data from registered shims for a user I have created, e.g. I cannot see connect buttons for Fitbit and Withings.

So what is my best strategy here. Can I go back to the v0.4.5 tag and copy the new implementations of the shims in and have it work? Or is there more work here than I am thinking? Can I just copy the old /registry handler and have it start working in the new code? Or is there going to be a commit soon that will make everything work?

I am willing to put in some programming effort because we need Shimmer for our data pilot. I just need to know what to do.

Best,
-Keith

**Comment by emersonf** [comment]
@kmhughes the [release notes for 0.5.0](https://github.com/openmhealth/shimmer/releases/tag/v0.5.0) state that

```The console hasn't been updated to work with version 0.5.0 yet. Postman is the preferred way to visually exercise the API until it is updated.```

While we intend to restore the console's functionality, we have more pressing updates on our plate for Shimmer, specifically re: getting other shims to use the newer schemas. 

If Postman isn't good enough for what you're trying to do, I'll need to understand what you're trying to use the console for to help you further.

**Comment by kmhughes** [comment]
@emersonf Ahh, very much my bad. I apologize. I was going through the README which still talks about using the console, so I unfortunately assumed it still worked. I will very definitely keep an eye on the release notes going forward now that I know to look there.

I actually figured out that I could still authenticate and access data before you posted this response. I had looked at the internal APIs and could see why the console didn't work, but there was no clear path on how to make it work, so I tried just using the authorize REST endpoints and found I was able to access data. I am writing code that accesses your API to get the data we want on a schedule and I can move forward with that code now. Postman will not help us with our project, but it is good to know it is there.

I tried using some of the intradaily settings with a server I built myself and got some errors when accessing some of the Fitbit data (steps and heartrate). I will check myself on how the schema is brought in, update it if necessary, and see if I still get those errors. I don't think we care about the intradaily information any more, but I just want to know I can use it if needed.

**Comment by emersonf** [comment]
Glad you can proceed.

We should have update the README, so I'll keep this issue open until we either update the docs or restore the console functionality.

**Comment by HeenaAgg** [comment]
Hi @kmhughes,

I m also trying to make Shimmer running. I am trying to authorize endpoints but I m unable to, redirect URL which i got back is empty and access parameters inserted into db does not contain client id and client secret, access token...these values are null.
 
Can you please guide how u make it running, what steps you followed.

Thanks

**Comment by drowhunter** [comment]
Any updates on when the console will be updated. Its been a minute.

**Comment by drowhunter** [comment]
> @emersonf Ahh, very much my bad. I apologize. I was going through the README which still talks about using the console, so I unfortunately assumed it still worked. I will very definitely keep an eye on the release notes going forward now that I know to look there.
> 
> I actually figured out that I could still authenticate and access data before you posted this response. I had looked at the internal APIs and could see why the console didn't work, but there was no clear path on how to make it work, so I tried just using the authorize REST endpoints and found I was able to access data. I am writing code that accesses your API to get the data we want on a schedule and I can move forward with that code now. Postman will not help us with our project, but it is good to know it is there.
> 
> I tried using some of the intradaily settings with a server I built myself and got some errors when accessing some of the Fitbit data (steps and heartrate). I will check myself on how the schema is brought in, update it if necessary, and see if I still get those errors. I don't think we care about the intradaily information any more, but I just want to know I can use it if needed.

I am we are having the same issue could you explain in more detail how you worked around it?

**Comment by kmhughes** [comment]
I ended up writing some code that does an AJAX call to the /authorize endpoint of Shimmer when my web application is called (the server side of my app does the AJAX call to Shimmer), took the response apart to get the URL to send the user to to initiate the OAUTH. It's part of a piece of somewhat complex code, so I can't really share it.


Unless I am mistaken, the Shimmer project seems stalled out. I forked the project to get some changes done that I needed, for example, getting a real JSON response and not just a 500 response code if the username is not authorized.

**Comment by drowhunter** [comment]
Can I see the fork?

**Comment by kmhughes** [comment]
@drowhunter Unfortunately it is in a private gitlab repo, so I can't. I have considered periodically pushing a copy of the fork out, though if I do that I need to remove some stuff from the repo. There are some schemas I want to write that Shimmer doesn't support, add some shim endpoints not currently supported, etc. I thought about pushing patches to the main project, but since no one else is working on it, it seemed easiest just to fork and add in what I specifically need.

**Comment by drowhunter** [comment]
No problem I was able to get it working in the end.

**Comment by drowhunter** [comment]
I was somewhat concerned because it seems this project is on life support atm.

**Comment by kmhughes** [comment]
I agree, it seems on life support. Not committing to anything, but perhaps
we should all discuss what we want and see what kind of overlap we have.
